server/job.go

```diff
@@ -181,12 +182,12 @@ func (p *Plugin) DeleteJob(ctx context.Context, id string) error {
 		return nil
 	}
 
-	err = p.fileBackend.RemoveDirectory(filepath.Dir(jobs[id].DumpLocation))
-	if err != nil {
-		p.API.LogError("dump could not be deleted", "id", jobs[id].ID, "err", err.Error())
-	} else {
+	if ok, err2 := jobs[id].DeleteDump(p); err2 != nil {
+		p.API.LogError("dump could not be deleted", "id", jobs[id].ID, "err", err2.Error())
+	} else if ok {
 		p.API.LogDebug("dump deleted", "id", jobs[id].ID, "file", jobs[id].DumpLocation)
 	}
```

Comment:
Nitpick: is the new `error` variable really needed?

```suggestion
	ok, err := jobs[id].DeleteDump(p);
	if  err != nil {
		p.API.LogError("dump could not be deleted", "id", jobs[id].ID, "err", err.Error())
	} else if ok {
		p.API.LogDebug("dump deleted", "id", jobs[id].ID, "file", jobs[id].DumpLocation)
	}
```